packages/node_modules/@ciscospark/widget-recents/src/container.js

```diff
@@ -537,6 +594,12 @@ export class RecentsWidget extends Component {
     const isWebRTCSupported = media.getIn(['webRTC', 'isSupported']);
 
     if (spacesList && hasFetchedSpaces) {
+      const spacesListWithActivityText = spacesList.map((space) => {
+        const s = Object.assign({}, space);
+        s.activityText = this.constructActivityText(space);
+        return s;
+      });
```

Comment:
Seems like this, along with the `constructActivityText` are prime examples of code that can be a selector. What do you think?

Reply:
I started off making this a selector, but it renders a component and uses format message so I kept it in the main render process. Would like for it to not have to re-process every time, so I'm up for suggestions on how to do those within a selector. Maybe we can just memoize it somehow outside of the render.